src/validation/entities/schemas/__tests__/hex.schema.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { HexSchema } from '@/validation/entities/schemas/hex.schema';
+import { faker } from '@faker-js/faker';
+
+describe('HexSchema', () => {
+  it('should validate a valid hex string', () => {
+    const value = faker.string.hexadecimal();
+
+    const result = HexSchema.safeParse(value);
+
+    expect(result.success).toBe(true);
+    if (!result.success) {
+      throw new Error('HexSchema failed to validate a valid hex string.');
+    }
```

Comment:
Are this kind of assertions (here and also in `address.schema.spec.ts` and `relay.dto.schema.spec.ts`) needed? I mean, if `result.success` is `false` Jest will inform of the failure.

Reply:
Yes. The types of `zod` are very strict, e.g.

```ts
type Result = { success: true, data: ... } | { success: false, error: ... }
```

They're needed to appease TypeScript:

![image](https://github.com/safe-global/safe-client-gateway/assets/20442784/d15d85d7-de81-456c-b06a-e17d81f25866)

Reply:
I see, thanks for the explanation. This is not a strong opinion at all, but wdyt about:
```javascript
expect(result.success && result.data).toBe(value);
```
Instead of:
```javascript
expect(result.success).toBe(true);
if (!result.success) {
  throw new Error('HexSchema failed to validate a valid hex string.');
}
expect(result.data).toBe(value);
```

I'm just saving some lines which it seems would get repeated in many tests, but this is absilutely nitpick, I'm fine with your code 👏🏻